riscv/src/parser.rs

```diff
@@ -36,11 +34,44 @@ impl Register {
     }
 }
 
+#[derive(Clone, Copy)]
+pub enum UnaryOpKind {
+    HiDataRef,
+    LoDataRef,
+    /// We capture parenthesis as an unary operation so that we can print it
```

Comment:
Hm, not sure if I would require that. It's an *abstract* syntax tree. I see no harm in adding parentheses everywhere in display just to ensure precedence is correct.

Reply:
Removed parethesis from AST, at expense of a lot of parenthesis in `Display`.